src/System.Text.Json/tests/JsonDocumentTests.cs

```diff
@@ -3034,6 +3045,124 @@ public static void EnsureResizeSucceeds()
+        {
+            JsonElement jElement = (JsonElement)JsonDocument.Parse(jsonString).RootElement.Clone();
+            Assert.False(jElement.ValueEquals(other));
+            Assert.False(jElement.ValueEquals(other.AsSpan()));
+            byte[] otherGetBytes = Encoding.UTF8.GetBytes(other);
+            Assert.False(jElement.ValueEquals(otherGetBytes));
+        }
+
+        [Theory]
+        [InlineData("\"hello\"", new char[1] { (char)0xDC01 })]    // low surrogate - invalid
+        [InlineData("\"hello\"", new char[1] { (char)0xD801 })]    // high surrogate - missing pair
+        public static void InvalidUTF16Search(string jsonString, char[] lookup)
+        {
+            JsonElement jElement = (JsonElement)JsonDocument.Parse(jsonString).RootElement.Clone();
+            Assert.False(jElement.ValueEquals(lookup));
+        }
+
+        [Theory]
+        [InlineData("\"connectionId\"", "\"conne\\u0063tionId\"")]
+        [InlineData("\"conne\\u0063tionId\"", "bonnectionId")] // intentionally changing the expected starting character
+        public static void ValueEquals_JsonTokenStringType_False(string jsonString, string otherText)
+        {
+            JsonElement jElement = (JsonElement)JsonDocument.Parse(jsonString).RootElement.Clone();
+            Assert.False(jElement.ValueEquals(otherText));
+            Assert.False(jElement.ValueEquals(otherText.AsSpan()));
+            byte[] expectedGetBytes = Encoding.UTF8.GetBytes(otherText);
+            Assert.False(jElement.ValueEquals(expectedGetBytes));
+        }
+
+        [Theory]
+        [InlineData("{}")]
+        [InlineData("{\"\" : \"\"}")]
+        [InlineData("{\"sample\" : \"\"}")]
+        [InlineData("{\"sample\" : null}")]
+        [InlineData("{\"sample\" : \"sample-value\"}")]
+        [InlineData("{\"connectionId\" : \"123\"}")]
+        public static void ValueEquals_JsonTokenNotString_Success(string jsonString)
+        {
+            const string errorMessage = "The requested operation requires an element of type 'String', but the target element has type 'Object'.";
+            JsonElement jElement = (JsonElement)JsonDocument.Parse(jsonString).RootElement.Clone();
+            AssertExtensions.Throws<InvalidOperationException>(() => jElement.ValueEquals("throws-anyway"), errorMessage);
+            AssertExtensions.Throws<InvalidOperationException>(() => jElement.ValueEquals("throws-anyway".AsSpan()), errorMessage);
+            byte[] expectedGetBytes = Encoding.UTF8.GetBytes("throws-anyway");
+            AssertExtensions.Throws<InvalidOperationException>(() => jElement.ValueEquals(expectedGetBytes), errorMessage);
+        }
+
+        [Fact]
+        public static void NameEquals_Default_True()
+        {
+            string s_object = $"{{ \"aPropertyName\" : \"itsValue\" }}";
+            var Object = JsonDocument.Parse(s_object).RootElement.Clone();
```

Comment:
```suggestion
            JsonElement element = JsonDocument.Parse(s_object).RootElement.Clone();
```

(Though, the line needs to be rewritten to use a using, and not use clone.)